### PR TITLE
Introduce zlo into hopper

### PR DESCRIPTION
**Pull request by blt**
This new serializer is a fork of bincode which reduces the size of
the on-disk payloads for us. It looks to be about 50% for the
cernan use-case.

Signed-off-by: Brian L. Troutwine <blt@postmates.com>

### CODE REVIEW

**Review by pulltab — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -161,7 +163,7 @@ where
     let sz = mem::size_of::<T>();
     let max_bytes = if max_bytes < sz { sz } else { max_bytes };
     let fs_lock = sync::Arc::new(sync::Mutex::new(private::FsSync::new(cap)));
-    let sender = try!(Sender::new(name, &snd_root, max_bytes, fs_lock.clone()));
+    let sender = try!(Sender::new(name, &snd_root, max_bytes, Arc::clone(&fs_lock)));
```
Comment (gliush) [inline src/lib.rs:166]: Not sure, why you have done this.
According to https://doc.rust-lang.org/std/sync/struct.Arc.html
these two syntaxes are equivalent

File: src/sender.rs [hunk 2]
```diff
@@ -32,11 +31,12 @@ where
     T: Serialize + Deserialize<'de>,
 {
     fn clone(&self) -> Sender<T> {
+        use std::sync::Arc;
```
Comment (gliush) [inline src/sender.rs:34]: If we use previous syntax, all these `use std::sync::Arc` will be not needed

**Review by gliush — COMMENTED** [review]
LGTM, though, I'd use the previous "Arc clone" syntax.

File: src/lib.rs (hunk 1 above)
Comment (blt) [reply src/lib.rs:166]: Clippy recommendation. https://github.com/rust-lang-nursery/rust-clippy/issues/1645

File: src/sender.rs (hunk 2 above)
Comment (blt) [reply src/sender.rs:34]: We could also just `use std::sync` at a global level instead. No compelling reason to do this over the global import.

**Pull request merged**